pkg/objstore/objstore.go

```diff
@@ -113,17 +114,19 @@ func DownloadFile(ctx context.Context, bkt BucketReader, src, dst string) error
 	if err != nil {
 		return errors.Wrap(err, "get file")
 	}
-	defer rc.Close()
+	defer runutil.LogOnErr(nil, rc, "download block's file reader")
 
 	f, err := os.Create(dst)
 	if err != nil {
 		return errors.Wrap(err, "create file")
 	}
 	defer func() {
-		f.Close()
+		runutil.LogOnErr(nil, f, "download block's output file")
 		// Best-effort cleanup.
 		if err != nil {
-			os.Remove(dst)
+			if rerr := os.Remove(dst); rerr != nil {
+				fmt.Println("failed to best effortly remove ", dst, "err:", rerr)
```

Comment:
`fmt` ... why not pass logger into these funcs?

Reply:
Another thought ... do we want to add metrics around the failure to clean up?

Reply:
just wanted to avoid additional parameter, but I guess it might be necessary

Reply:
regarding metric.. I am not sure. I would keep metrics alert on bad symptomps, like "I cannot run further because I have some malformed dir". However most of processes are working on tmp dirs and removed on different levels. So this removal is only "nice" thing to have, nothing critical. 

We definitely need to rething retry path and how to handle these. The issue https://github.com/improbable-eng/thanos/issues/318 is for it.

Reply:
What about being SIMPLE here and be clear that the directory is malformed and can have partial download and caller should be responsible for clean up?

Reply:
Yeah, we should be using logger instead of print statement. Also you can remove `"best effortly"` substring.
Also add logger to other places too as we will eventually need it anyways as the program grows.